### PR TITLE
Add GIGS tests.

### PR DESCRIPTION
**Pull request by micahcochran**
This is to implement #374 

This is a Python framework that uses JSON formatted test files.  It needs pyproj to be installed.

It does 2 different kinds of tests.  A simple conversion test and 1,000 roundtrips.

**Notes:**
~~\* Python 3 seems to pass more of the tests, so I assume that it is more precises, where Python 2.7 seems to have some slight rounding errors with the 1,000 roundtrip tests.  I assume that this is some issue with Python or pyproj.  So, Python 3.3+ is recommended for testing. :smiley:    This was a very recent discovery.~~
- Some of the tests fail and they are not included in Travis.  Still they are in the commit for testing.  The [TESTNOTES.md](https://github.com/micahcochran/proj.4/blob/py-gigs-test4/test/gigs/TESTNOTES.md) notes how badly they were off.  :arrow_backward: that needs to be reformatted.
- There could still be issues with the testing methods itself.  GIGS uses Degrees Minutes Seconds for lat/long, where I've used Decimal Degrees for lat/long.
- Using cs2cs was attempted, but there were some precision errors and it is a few orders of magnitude slower.  (It seems like a few thousand Python Popen calls per test is expensive.)  18 minute runtime for cs2cs testing, versus the 10 second runtime for pyproj.  The cs2cs code is still in the code. 

I'd like this to be reviewed at least a little bit before merging, so **this is not yet ready for merging**.

I have the question:  Should this be a test suite/framework that is outside of Proj.4 project?  BUT I need to ask the International Association of Oil & Gas Producers how their material can be reproduced appropriately.

**EDIT**: Python 2.x & 3.x now work the same for testing purposes.  This was due to the zip function being different in Python 3.x.  I've worked around this.

### CODE REVIEW

**Comment by QuLogic** [comment]
Can you set up the JSON so that they are all formatted the same way? Sometimes the coordinates are on multiple lines, and sometimes all on one (very long) line.

File: test/gigs/TESTNOTES.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+ * pyproj (optional but highly recommended), this speeds up tests, makes
+    results more precise but has the trade-off of making installation a
+    little more compilated
+
+
+
+### Installing pyproj ##################################################
+
+ * Install `pip` if not installed, should already be installed on new versions 
+      (Python 2 >=2.7.9 or Python 3 >=3.4) or if using a virtual environment for python.
```
Comment (QuLogic) [inline test/gigs/TESTNOTES.md]: I thought you recommended 3.3?

File: test/gigs/TESTNOTES.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+```
+         $ pip install cython
+```
+    - install latest release
+        - set PROJ_DIR environment variable to an installed version of PROJ4
+          library.  This should have the directories include/ lib/ & 
+          share/proj/ underneath it.
+    - Linux  (default ./configure)
+```
+         $ sudo PROJ_DIR=/usr/local pip install pyproj
```
Comment (QuLogic) [inline test/gigs/TESTNOTES.md]: `sudo` should not be encouraged, especially if using a virtual environment.

**Comment by micahcochran** [comment]
Thanks @QuLogic . I have reformatted the JSON files, but I will admit they aren't perfect.  I've removed `sudo` from the TESTNOTES example.  I fixed a bug that make Python 2.x & 3.x work the same.

**Comment by QuLogic** [comment]
I suggest running `pep8` on the Python files to check for consistent formatting.

**Comment by micahcochran** [comment]
Thanks @QuLogic. I updated the formatting based on `flake8` and fixed most of the problems.  I only left some of the greater than 79 character lines.

I just sent an email to IOGP about this PR.

**Comment by micahcochran** [comment]
I have given up on getting a response from IOGP.  I am considering creating a separate project, so that this project won't have to deal with the licensing issues of for using test data that is IOGP property.  Also, this test framework could be useful for other coordinate transformation libraries.

Any suggestions?

**Pull request merged**